### ISSUE TITLE
Pass parameters to `statsmodels.lowess` in a call to `regplot`

### ISSUE THREAD

**Issue by RuolinZheng08** [mwaskom/seaborn#2351]
This feature might have been requested before: May we customize the `lowess` parameter in `regplot` and pass arguments to `statsmodels` function `nonparametric.smoothers_lowess.lowess(endog, exog, frac=0.6666666666666666, it=3, delta=0.0, xvals=None, is_sorted=False, missing='drop', return_sorted=True)`?

I saw the mention of a `seaborn.lowessplot` function mentioned in #607. Is there any plan to implement that?

Thanks for your hard work on this beautiful visualization library.

**Comment by mwaskom** [comment]
Yes, I would like to expose an interface for LOWESS-based regression plots that allows more control over the smoothing parameters. I'm also interested in other enhancements, like using statsmodel's new GAM tools for faster local smoothing on large datasets (similar to what ggplot does) and getting confidence intervals on the prediction. I continue to think the right way forward is to separate out the nonparametric regressionss into their own function. This is tentatively on the roadmap for a 0.12 release. It's not something that should happen before the regression module can be refactored to use the new core data infrastructure introduced in 0.11.

**Comment by leoluecken** [comment]
I also needed this and did a first draft for the implementation. See PR to follow.
